src/warehouses/components/WarehouseZones/WarehouseZones.tsx

```diff
@@ -28,16 +36,108 @@ const useStyles = makeStyles(
 
 const WarehouseInfo: React.FC<WarehouseInfoProps> = ({
   zones,
-  onShippingZoneClick
+  data,
+  onShippingZoneClick,
+  onChange
 }) => {
   const classes = useStyles({});
   const intl = useIntl();
 
+  const isPrivateChoices = [
+    {
+      label: (
+        <>
+          <FormattedMessage defaultMessage="Private Stock" />
+          <Typography variant="caption" color="textSecondary">
+            <FormattedMessage defaultMessage="If enabled stock in this warehouse won't be shown" />
+          </Typography>
+          <FormSpacer />
+        </>
+      ),
+      value: "true"
+    },
+    {
+      label: (
+        <>
+          <FormattedMessage defaultMessage="Public Stock" />
+          <Typography variant="caption" color="textSecondary">
+            <FormattedMessage defaultMessage="If enabled stock in this warehouse will be shown" />
+          </Typography>
+        </>
+      ),
+      value: "false"
+    }
+  ];
+
+  const clickAndCollectChoices = [
+    {
+      label: (
+        <>
+          <FormattedMessage defaultMessage="Disabled" />
+          <Typography variant="caption" color="textSecondary">
+            <FormattedMessage defaultMessage="If selected customer won't be able to choose this warehouse as pickup point" />
+          </Typography>
+          <FormSpacer />
+        </>
+      ),
+      value: WarehouseClickAndCollectOptionEnum.DISABLED
+    },
+    {
+      label: (
+        <>
+          <FormattedMessage defaultMessage="Local stock only" />
+          <Typography variant="caption" color="textSecondary">
+            <FormattedMessage defaultMessage="If selected customer will be able to choose this warehouse as pickup point. Ordered products will be only fulfilled from this warehouse stock" />
+          </Typography>
+          <FormSpacer />
+        </>
+      ),
+      value: WarehouseClickAndCollectOptionEnum.LOCAL
+    },
+    {
+      label: (
+        <>
+          <FormattedMessage defaultMessage="All warehouses" />
+          <Typography variant="caption" color="textSecondary">
+            <FormattedMessage defaultMessage="If selected customer will be able to choose this warehouse as pickup point. Ordered products can be shipped here from a different warehouse" />
+          </Typography>
+        </>
+      ),
+      value: WarehouseClickAndCollectOptionEnum.ALL
+    }
+  ];
+
+  const clickAndCollectChoicesPrivate = [
```

Comment:
Isn't this one the same as `clickAndCollectChoices` but without one option?
If so I'd prefer `clickAndCollectChoices` defines as `const clickAndCollectChoices = [...clickAndCollectChoicesPrivate, { ...publicOption }` to improve readability and reusability of options.

Reply:
Yes, as we spoke, I followed your advice and I have used `filter`